great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -4841,7 +4831,7 @@ def _instantiate_datasource_from_config_and_update_project_config(
         """
         # If attempting to override an existing value, ensure that the id persists
         name = config.name
-        if not config.id and name in self.datasources:
+        if not config.id and name and name in self.datasources:
```

Comment:
If `name` is `None` or an empty string it won't find a match in `self.datasource`.

So why do we need the extra `and name` here?